# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [1]:
import os
if not os.path.exists("flyrank-ml-internship"):
    !git clone https://github.com/musab855/flyrank-ml-internship.git
%cd flyrank-ml-internship

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 115, done.
remote: Counting objects: 100% (115/115), done.
remote: Compressing objects: 100% (86/86), done.
remote: Total 115 (delta 31), reused 78 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (115/115), 1.84 MiB | 4.29 MiB/s, done.
Resolving deltas: 100% (31/31), done.
/content/flyrank-ml-internship


## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the **Refresh / Content Opportunity Scoring** lane because it focuses on helping content teams prioritize which pages should be reviewed first. Large websites often contain hundreds or thousands of pages, making it difficult to decide where to invest limited time and resources. By analyzing observed search performance and engagement signals, this project aims to identify pages that may benefit from a content review. I selected this lane because it combines practical business decision-making with machine learning and provides a strong foundation for building an intelligent decision-support system.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Question**: Which content pages should be prioritized for review based on their observed search performance and engagement signals?

**Decision**: Determine which content pages should be prioritized for review based on observed performance signals.

**Who acts?**: SEO specialists, content strategists, and website managers use these recommendations to plan content updates.

**Cost of a wrong recommendation**: If a low-priority page is selected, valuable time and resources may be wasted. If an important declining page is overlooked, the website may continue losing search visibility, traffic, and potential business opportunities.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print()

print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Average sessions (90d):", round(df["sessions_90d"].mean(), 2))
print()

print(df["trend_direction"].value_counts())


Rows: 30000
Columns: 44

Average impressions (90d): 5200.37
Average sessions (90d): 37.07

trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


Out of 30,000 pages, 16,262 (54%) show a declining trend, and average impressions (5,200) far exceed average sessions (37) — confirming a large visibility-to-traffic gap that manual review can't address at scale, making prioritization essential.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project is intended to provide **decision-support** by identifying observed patterns associated with content performance and recommending pages for review. The results will be **directional** and based on the available data rather than definitive predictions. The project **cannot** prove that refreshing a page will improve search performance, establish causal relationships, or predict Google's ranking algorithm. Final decisions should always be reviewed by human experts.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.